mrjob/util.py

```diff
@@ -611,7 +621,13 @@ def safeeval(expr, globals=None, locals=None):
     """
     # blank out builtins, but keep None, True, and False
     safe_globals = {'__builtins__': None, 'True': True, 'False': False,
-                    'None': None, 'set': set, 'xrange': xrange}
+                    'None': None, 'set': set}
+
+    # xrange is range in Python 3
+    if hasattr(__builtins__, 'xrange'):
+        safe_globals['xrange'] = xrange
+    else:
+        safe_globals['range'] = range
```

Comment:
Nah, the point of `safeeval()` is to decode reprs, which won't have `range` in them on Python 2, since in Python 2, `range()` is just a function that returns a list of numbers.

By the way turns out this code was wrong; `__builtins__` is actually a dictionary on Python 2 (it supports attribute access, but not `hasattr()`). Good thing I added a test case!